File: frontend/src/app/models/reservation.ts
export type ReservationStatus = 'pending' | 'completed' | 'cancelled';

export interface Reservation {
  id: number;
  listingId: number;
  listingTitle: string;
  establishmentName: string;
  pickupTime: string;
  status: ReservationStatus;
  reservedAt: string;
}

export interface BusinessReservation {
  id: number;
  listingId: number;
  listingTitle: string;
  customerName: string;
  customerEmail: string;
  pickupTime: string;
  status: ReservationStatus;
  reservedAt: string;
}

File: backend-python/app/schemas/reservation.py
from datetime import datetime

from app.models.reservation import ReservationStatus
from sqlmodel import SQLModel


class ReservationCreate(SQLModel):
    listing_id: int


class ReservationRead(SQLModel):
    id: int
    listingId: int
    listingTitle: str
    establishmentName: str
    pickupTime: str
    status: ReservationStatus
    reservedAt: datetime


class BusinessReservationRead(SQLModel):
    id: int
    listingId: int
    listingTitle: str
    customerName: str
    customerEmail: str
    reservedAt: datetime
    status: ReservationStatus
    pickupTime: str


class ReservationStatusUpdate(SQLModel):
    status: ReservationStatus
